### PR TITLE
add feat edit profile

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
[- ticket redmine
](https://edu-redmine.sun-asterisk.vn/issues/75203)
## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/config/cloudinary.config.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {v2 as cloudinary} from 'cloudinary';
+          
+  cloudinary.config({ 
+  cloud_name: 'dnn7xd45c', 
+  api_key: '312742596612911', 
+  api_secret: '2KluGSQA1pzd5XxmMF4lV8REgeo',
```
Comment (huongnt-2545) [inline src/config/cloudinary.config.ts]: cho giá trị nhạy cảm vào env e nha

File: src/controller/user.controller.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import Swal from 'sweetalert2';
+import { SEX_USER } from '../constant/enum';
+import { UserChangePassword, UserUpdateDto } from '../dto/user/user.dto';
+import { plainToClass } from 'class-transformer';
+import { validate } from 'class-validator';
+import { checkPassword } from '../utils/auth/auth';
+
+const userService = new UserService();
+
+export const GetProfileUser = async (req: Request, res: Response) => {
```
Comment (huongnt-2545) [inline src/controller/user.controller.ts]:
```suggestion
export const getProfileUser = async (req: Request, res: Response) => {
```

File: src/controller/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+export const uploadCloudinary = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const file = req.file;
+    if (file) {
+      const result = await cloudinary.uploader.upload(file?.path || '', {
+        folder: 'morii',
```
Comment (huongnt-2545) [inline src/controller/user.controller.ts]: giá trị folder path em khai báo trong file constant nha

File: src/controller/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      if (result?.url) req.body.avatar = result?.url;
+    }
+    next();
+  } catch (error) {
+    req.flash('error', 'user.not-upload-image');
+    res.render('user', { message: req.flash('error') });
+  }
+};
+
+export const UpdateProfileUser = async (req: Request, res: Response) => {
```
Comment (huongnt-2545) [inline src/controller/user.controller.ts]:
```suggestion
export const updateProfileUser = async (req: Request, res: Response) => {
```

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Comment by huongnt-2545** [comment]
merged

**Pull request merged**